### PR TITLE
Default to using more reliable virt detection

### PR DESCRIPTION
**Pull request by nkukard**
dmidecode & dmesg is not terribly reliable to detect many virtualized systems....

  root@development:/etc# salt-call grains.items  
    virtual: physical
    virtual_subtype: LXC

Rather default to more reliable techniques.

Also, only enable lspci should we plan to use it and also if the directory exists.

### CODE REVIEW

**Comment by jfindlay** [comment]
@nkukard, there are some [lint errors](https://jenkins.saltstack.com/job/salt-pr-lint-n/8927/violations/).

**Comment by nkukard** [comment]
@jfindlay, This should fix it :)

**Comment by jfindlay** [comment]
@nkukard, sorry, just [one more](https://jenkins.saltstack.com/job/salt-pr-lint-n/8930/violations/file/salt/grains/core.py/).  I don't know why that didn't get picked up the first time.

**Comment by nkukard** [comment]
@jfindlay hopefully this sorts it out :)

File: salt/grains/core.py [hunk 1]
```diff
@@ -492,16 +492,15 @@ def _virtual(osdata):
...
-            _cmds = ['systemd-detect-virt', 'virt-what', 'dmidecode', 'dmesg']
-        else:
-            _cmds = ['systemd-detect-virt', 'virt-what', 'dmidecode', 'lspci',
-                     'dmesg']
+        if os.path.exists('/proc/bus/pci'):
+            _cmds += ['lspci']
+
+    # Add additional last resort commands
+    if osdata['kernel'] not in skip_cmds:
+        _cmds += ['dmesg'];
```
Comment (cachedout) [inline salt/grains/core.py]: Please remove this semi-colon and then we can get this merged. Thanks!

**Comment by nkukard** [comment]
take 99999 , seems I forgot to push :(

**Comment by cachedout** [comment]
No worries. We'll get this merged today. Many thanks @nkukard

**Pull request merged**